pkg/k8sutil/k8sutil.go

```diff
@@ -133,6 +133,7 @@ func CreateOrUpdateService(sclient clientv1.ServiceInterface, svc *v1.Service) e
 		}
 	} else {
 		svc.ResourceVersion = service.ResourceVersion
+		svc.SetOwnerReferences(mergeOwnerReferences(service.GetOwnerReferences(), svc.GetOwnerReferences()))
```

Comment:
This way we're never pruning non-existant prometheus/alertmanager owner references right? I think we should.

Reply:
@brancz An undocumented feature of Kubernetes GC is that it will actually remove the `ownerReference` from the child object once it is deleted, so we don't need to do this ourselves. We actually have a [test for this in OLM](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/gc_e2e_test.go#L22).

Reply:
Woah, I had no idea. Thanks for sharing!